src/CSharp/CodeCracker/Usage/DisposableVariableNotDisposedCodeFixProvider.cs

```diff
@@ -69,6 +101,38 @@ public static SyntaxNode CreateUsing(SyntaxNode root, ObjectCreationExpressionSy
             return newRoot;
         }
 
+        private static string GetIdentifierName(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel)
+        {
+            var identifierName = "disposableObject";
+
+            var type = objectCreation.Type;
+            if (type.IsKind(SyntaxKind.QualifiedName))
+            {
+                var name = (QualifiedNameSyntax)type;
+                identifierName = LoverCaseFirstLetter(name.Right.Identifier.ValueText);
+            }
+            else if (type is SimpleNameSyntax)
+            {
+                var name = (SimpleNameSyntax)type;
+                identifierName = LoverCaseFirstLetter(name.Identifier.ValueText);
+            }
+
+            var confilctingNames = from symbol in semanticModel.LookupSymbols(objectCreation.SpanStart)
+                                   let symbolIdentifierName = symbol?.ToDisplayParts().LastOrDefault(IncludeOnlyPartsThatAreName).ToString()
+                                   where symbolIdentifierName != null && symbolIdentifierName.StartsWith(identifierName)
+                                   select symbolIdentifierName;
+
+            var identifierPostFix = 0;
+            while (confilctingNames.Any(p => p == identifierName + ++identifierPostFix)) { }
+            return identifierName + (identifierPostFix == 0 ? "" : identifierPostFix.ToString());
+        }
+
+        private static string LoverCaseFirstLetter(string name) => char.ToLowerInvariant(name[0]) + name.Substring(1);
+        public static bool IncludeOnlyPartsThatAreName(SymbolDisplayPart displayPart) =>
```

Comment:
Also, perhaps it should be named "LowerCaseFirstLetter" instead of "LoverCaseFirstLetter". While we like love our code, that may be going a bit too far ;-)